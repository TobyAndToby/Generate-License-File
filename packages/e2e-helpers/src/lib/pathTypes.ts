import path from "path";

type DescribeCallback = (packageJsonPath: string) => void;

export const describeRelativeAndAbsolutePaths = (
  relativePackageJsonPath: string,
  callback: DescribeCallback,
) => {
  describe("for a relative path", () => {
    callback(relativePackageJsonPath);
  });

  const absolutePackageJsonPath = path.join(process.cwd(), relativePackageJsonPath);

  describe("for an absolute path", () => {
    callback(absolutePackageJsonPath);
  });
};
