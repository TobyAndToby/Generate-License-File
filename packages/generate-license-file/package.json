{
  "name": "generate-license-file",
  "version": "0.0.0",
  "description": "Generates a text file containing all of the licenses for your production dependencies",
  "main": "src/index.js",
  "license": "ISC",
  "type": "commonjs",
  "contributors": [
    {
      "name": "Toby Smith",
      "url": "https://github.com/tobysmith568"
    },
    {
      "name": "Toby Bessant",
      "url": "https://github.com/tobybessant"
    }
  ],
  "files": [
    "src",
    "bin"
  ],
  "keywords": [
    "cli",
    "license",
    "licence",
    "third",
    "party",
    "file",
    "generate",
    "oss"
  ],
  "repository": {
    "type": "git",
    "url": "git+https://github.com/TobyAndToby/generate-license-file.git"
  },
  "bugs": {
    "url": "https://github.com/TobyAndToby/generate-license-file/issues"
  },
  "homepage": "https://generate-license-file.js.org",
  "bin": {
    "generate-license-file": "bin/generate-license-file"
  }
}
