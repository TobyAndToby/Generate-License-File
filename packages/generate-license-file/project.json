{
  "name": "generate-license-file",
  "$schema": "../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "packages/generate-license-file/src",
  "projectType": "library",
  "targets": {
    "build": {
      "executor": "@nx/js:tsc",
      "outputs": ["{options.outputPath}"],
      "options": {
        "outputPath": "dist/packages/generate-license-file",
        "main": "packages/generate-license-file/src/index.ts",
        "tsConfig": "packages/generate-license-file/tsconfig.lib.json",
        "assets": [
          "packages/generate-license-file/*.md",
          "packages/generate-license-file/bin/**",
          "LICENSE.md"
        ],
        "buildableProjectDepsInPackageJsonType": "dependencies"
      }
    },
    "run": {
      "executor": "nx:run-commands",
      "options": {
        "command": "node ./dist/packages/generate-license-file/bin/generate-license-file"
      }
    },
    "version": {
      "executor": "nx:run-commands",
      "options": {
        "command": "node tools/scripts/version.mjs generate-license-file {args.version}"
      }
    },
    "publish": {
      "executor": "nx:run-commands",
      "options": {
        "command": "node tools/scripts/publish.mjs generate-license-file {args.tag}"
      }
    },
    "lint": {
      "executor": "@nx/linter:eslint",
      "outputs": ["{options.outputFile}"],
      "options": {
        "lintFilePatterns": ["packages/generate-license-file/**/*.ts"],
        "maxWarnings": 0
      }
    },
    "test": {
      "executor": "@nx/jest:jest",
      "outputs": ["{workspaceRoot}/coverage/{projectRoot}"],
      "options": {
        "jestConfig": "packages/generate-license-file/jest.config.ts",
        "passWithNoTests": false
      }
    }
  },
  "tags": []
}
