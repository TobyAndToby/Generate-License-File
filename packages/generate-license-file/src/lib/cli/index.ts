import { mainCommand } from "./commands/main";

export const main = async (args: string[]): Promise<void> => {
  await program.parseAsync(args);
};

const program = mainCommand;
