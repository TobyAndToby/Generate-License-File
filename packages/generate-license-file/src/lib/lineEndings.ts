import os from "os";

export const lineEndings = ["crlf", "lf"] as const;

/**
 * Used to specify which line endings to use in the generated file.
 *
 * Possible values are:
 *
 * `crlf` (`\r\n`)
 *
 * `lf` (`\n`)
 */
export type LineEnding = (typeof lineEndings)[number];

const lineEndingsMap = {
  crlf: "\r\n",
  lf: "\n",
} as const satisfies Record<LineEnding, string>;

export type LineEndingCharacters = (typeof lineEndingsMap)[LineEnding];
export const lineEndingCharacters = Object.values(lineEndingsMap);

export const isLineEnding = (input: unknown): input is LineEnding =>
  lineEndings.includes(input as LineEnding);

export const getLineEndingCharacters = (input: LineEnding | undefined): LineEndingCharacters => {
  if (input === undefined) {
    return os.EOL as LineEndingCharacters;
  }

  return lineEndingsMap[input];
};
