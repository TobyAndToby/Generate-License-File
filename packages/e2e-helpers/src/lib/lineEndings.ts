import { LineEnding } from "generate-license-file";

type DescribeEachLineEndingCallback = (lineEnding: LineEnding, lineEndingLiteral: string) => void;

const lineEndings: Record<LineEnding, string> = {
  crlf: "\r\n",
  lf: "\n",
};

export const describeEachLineEnding = (callback: DescribeEachLineEndingCallback) => {
  for (const key in lineEndings) {
    const lineEnding = key as LineEnding;
    const value = lineEndings[lineEnding];

    describe("for line ending " + key, () => callback(lineEnding, value));
  }
};
