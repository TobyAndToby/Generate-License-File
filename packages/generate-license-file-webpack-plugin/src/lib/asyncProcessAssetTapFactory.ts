import { getLicenseFileText, GetLicenseFileTextOptions } from "generate-license-file";
import { join } from "path/posix";
import { Compilation, Compiler, WebpackError } from "webpack";
import { LicenseFilePlugin } from "./licenseFilePlugin";
import { devImplementation } from "./devImplementation";
import { Options } from "./options";

export type AssetProcessingAsyncTap = (_: unknown, resolve: (error?: WebpackError) => void) => void;

const unknownError = "Unknown Error! Check for error output above.";

export const asyncProcessAssetTapFactory = (
  options: Options,
  compiler: Compiler,
  compilation: Compilation,
): AssetProcessingAsyncTap => {
  const pluginName = LicenseFilePlugin.name;

  const {
    outputFileName,
    outputFolder,
    isDev,
    pathToPackageJson,
    lineEnding,
    append,
    exclude,
    replace,
  } = options;

  const getLicenseFileTextOptions: GetLicenseFileTextOptions = {
    lineEnding,
    append,
    exclude,
    replace,
  };

  const RawSource = compiler.webpack.sources.RawSource;

  return (_, resolve) => {
    const implementation = resolveImplementation(isDev);

    implementation(pathToPackageJson, getLicenseFileTextOptions)
      .then(text => {
        const outputPath = join(outputFolder, outputFileName);
        compilation.emitAsset(outputPath, new RawSource(text));
        resolve();
      })
      .catch(error => {
        const errorMessage = `${pluginName}: ${error ?? unknownError}`;
        const webpackError = new WebpackError(errorMessage);
        compilation.errors.push(webpackError);
        resolve(webpackError);
      });
  };
};

const resolveImplementation = (isDev: boolean): typeof getLicenseFileText => {
  if (isDev) {
    return devImplementation;
  }

  return getLicenseFileText;
};
