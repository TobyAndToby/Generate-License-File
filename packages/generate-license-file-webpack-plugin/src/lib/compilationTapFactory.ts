import { Compilation, Compiler } from "webpack";
import { LicenseFilePlugin } from "./licenseFilePlugin";
import { asyncProcessAssetTapFactory } from "./asyncProcessAssetTapFactory";
import { Options } from "./options";

export type CompilationTap = (compilation: Compilation) => void;

export const compilationTapFactory = (options: Options, compiler: Compiler): CompilationTap => {
  const pluginName = LicenseFilePlugin.name;

  return compilation => {
    const processAssetTap = asyncProcessAssetTapFactory(options, compiler, compilation);

    const assetOptions = {
      name: pluginName,
      stage: Compilation.PROCESS_ASSETS_STAGE_ADDITIONAL,
    };

    compilation.hooks.processAssets.tapAsync(assetOptions, processAssetTap);
  };
};
