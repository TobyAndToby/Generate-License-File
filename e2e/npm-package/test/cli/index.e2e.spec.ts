import {
  describeEachLineEnding,
  describeRelativeAndAbsolutePaths,
} from "@generate-license-file/e2e-helpers";
import { exec } from "child_process";
import fs from "fs/promises";
import { LineEnding } from "generate-license-file";
import { lineEndings } from "generate-license-file/src/lib/lineEndings";
import path from "path";
import { promisify } from "util";

const execAsync = promisify(exec);

describe("cli", () => {
  describeRelativeAndAbsolutePaths("./package.json", (packageJsonPath) => {
    let outputFileName = "";

    let input = "";
    let output = "";

    let requiresUnlink = true;

    beforeEach(() => {
      outputFileName = `test-${Math.floor(Math.random() * 1000)}.txt`;

      input = packageJsonPath;
      output = path.dirname(packageJsonPath) + "/" + outputFileName;

      requiresUnlink = true;
    });

    afterEach(async () => {
      if (requiresUnlink) {
        await fs.unlink(output);
      }
    });

    it("should match snapshot with just --input and --output", async () => {
      await execAsync(
        `npx generate-license-file --input ${input} --output ${output}`
      );

      const result = await fs.readFile(output, "utf8");
      expect(result).toMatchSnapshot();
    });

    it("should match snapshot with --output given before --input", async () => {
      await execAsync(
        `npx generate-license-file --output ${output} --input ${input}`
      );

      const result = await fs.readFile(output, "utf8");
      expect(result).toMatchSnapshot();
    });

    it("should match snapshot using -i and -o", async () => {
      await execAsync(`npx generate-license-file -i ${input} -o ${output}`);

      const result = await fs.readFile(output, "utf8");
      expect(result).toMatchSnapshot();
    });

    it("should match snapshot with --input and --output wrapped in double quotes", async () => {
      await execAsync(
        `npx generate-license-file --input "${input}" --output "${output}"`
      );

      const result = await fs.readFile(output, "utf8");
      expect(result).toMatchSnapshot();
    });

    describe("overwrite", () => {
      it("should overwrite the file if --overwrite is given", async () => {
        const fileContent = "File content to be overwritten";

        await fs.writeFile(output, fileContent);

        await execAsync(
          `npx generate-license-file --input ${input} --output ${output} --overwrite`
        );

        const result = await fs.readFile(output, "utf8");
        expect(result).not.toMatch(fileContent);
      });
    });

    describeEachLineEnding((lineEnding, lineEndingLiteral) => {
      let lineEndingsNotUnderTest: LineEnding[] = [];

      beforeEach(async () => {
        lineEndingsNotUnderTest = lineEndings.filter((x) => x !== lineEnding);
      });

      it("should match snapshot when --eol is given", async () => {
        await execAsync(
          `npx generate-license-file --input ${input} --output ${output} --eol ${lineEnding}`
        );

        const result = await fs.readFile(output, "utf8");
        expect(result).toMatchSnapshot();
      });

      it("should contain the correct line ending value", async () => {
        const expectedLineEndingValue = lineEndingLiteral;

        await execAsync(
          `npx generate-license-file --input ${input} --output ${output} --eol ${lineEnding}`
        );

        const result = await fs.readFile(output, "utf8");
        expect(result).toContain(`package!${expectedLineEndingValue}https://`);
      });

      lineEndingsNotUnderTest.forEach((otherLineEnding) =>
        it(`should not contain the incorrect line ending value (${otherLineEnding})`, async () => {
          const incorrectLineEndingValue = lineEndingLiteral;

          await execAsync(
            `npx generate-license-file --input ${input} --output ${output} --eol ${lineEnding}`
          );

          const result = await fs.readFile(output, "utf8");
          expect(result).not.toContain(
            `package!${incorrectLineEndingValue}https://`
          );
        })
      );
    });

    describe("no-spinner", () => {
      it("should match snapshot when --no-spinner is given", async () => {
        await execAsync(
          `npx generate-license-file --input ${input} --output ${output} --no-spinner`
        );

        const result = await fs.readFile(output, "utf8");
        expect(result).toMatchSnapshot();
      });
    });

    describe("ci", () => {
      it("should match snapshot when --ci is given", async () => {
        await execAsync(
          `npx generate-license-file --input ${input} --output ${output} --ci`
        );

        const result = await fs.readFile(output, "utf8");
        expect(result).toMatchSnapshot();
      });
    });

    describe("omit-versions", () => {
      it("should match snapshot when --omit-versions is given", async () => {
        await execAsync(
          `npx generate-license-file --input ${input} --output ${output} --omit-versions`
        );

        const result = await fs.readFile(output, "utf8");
        expect(result).toMatchSnapshot();
      });
    });

    describe("version", () => {
      beforeEach(() => (requiresUnlink = false));

      it("should not produce an output when --version is given", async () => {
        await execAsync(
          `npx generate-license-file --input ${input} --output ${output} --version`
        );

        await expect(fs.stat(output)).rejects.toThrow();
      });

      it("should not produce an output when -v is given", async () => {
        await execAsync(
          `npx generate-license-file -i ${input} -o ${output} -v`
        );

        await expect(fs.stat(output)).rejects.toThrow();
      });

      it("should return the current version when --version is given", async () => {
        const versionInPackageJson = "*";

        const { stdout } = await execAsync(
          `npx generate-license-file --version`
        );

        expect(stdout.trim()).toBe(`v${versionInPackageJson}`);
      });

      it("should return the current version when -v is given", async () => {
        const versionInPackageJson = "*";

        const { stdout } = await execAsync(`npx generate-license-file -v`);

        expect(stdout.trim()).toBe(`v${versionInPackageJson}`);
      });
    });
  });
});
