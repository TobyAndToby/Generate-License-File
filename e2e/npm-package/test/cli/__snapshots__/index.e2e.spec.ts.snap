// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cli for a relative path ci should match snapshot when --ci is given 1`] = `
"This file was generated with the generate-license-file npm package!
https://www.npmjs.com/package/generate-license-file

The following npm package may be included in this product:

 - dep-four@1.0.0

This package contains the following license and notice below:

# Dep Four

This license file is spelt \`LICENCE\`.
This license should be found.

-----------

The following npm package may be included in this product:

 - dep-one@1.0.0

This package contains the following license and notice below:

# Dep One

This license file is spelt \`LICENSE.md\`.
This license should be found.

-----------

The following npm package may be included in this product:

 - dep-three@1.0.0

This package contains the following license and notice below:

# Dep Three

This license file is spelt \`LICENSE\`.
This license should be found.

-----------

The following npm packages may be included in this product:

 - dep-two-duplicate@1.0.0
 - dep-two@1.0.0

These packages each contain the following license and notice below:

# Dep Two

This license file is spelt \`LICENCE.md\`.
This license should be found.

-----------

The following npm package may be included in this product:

 - dep-five@1.0.0

This package contains the following license and notice below:

MIT

-----------

This file was generated with the generate-license-file npm package!
https://www.npmjs.com/package/generate-license-file
"
`;

exports[`cli for a relative path for line ending crlf should match snapshot when --eol is given 1`] = `
"This file was generated with the generate-license-file npm package!
https://www.npmjs.com/package/generate-license-file

The following npm package may be included in this product:

 - dep-four@1.0.0

This package contains the following license and notice below:

# Dep Four

This license file is spelt \`LICENCE\`.
This license should be found.

-----------

The following npm package may be included in this product:

 - dep-one@1.0.0

This package contains the following license and notice below:

# Dep One

This license file is spelt \`LICENSE.md\`.
This license should be found.

-----------

The following npm package may be included in this product:

 - dep-three@1.0.0

This package contains the following license and notice below:

# Dep Three

This license file is spelt \`LICENSE\`.
This license should be found.

-----------

The following npm packages may be included in this product:

 - dep-two-duplicate@1.0.0
 - dep-two@1.0.0

These packages each contain the following license and notice below:

# Dep Two

This license file is spelt \`LICENCE.md\`.
This license should be found.

-----------

The following npm package may be included in this product:

 - dep-five@1.0.0

This package contains the following license and notice below:

MIT

-----------

This file was generated with the generate-license-file npm package!
https://www.npmjs.com/package/generate-license-file
"
`;

exports[`cli for a relative path for line ending lf should match snapshot when --eol is given 1`] = `
"This file was generated with the generate-license-file npm package!
https://www.npmjs.com/package/generate-license-file

The following npm package may be included in this product:

 - dep-four@1.0.0

This package contains the following license and notice below:

# Dep Four

This license file is spelt \`LICENCE\`.
This license should be found.

-----------

The following npm package may be included in this product:

 - dep-one@1.0.0

This package contains the following license and notice below:

# Dep One

This license file is spelt \`LICENSE.md\`.
This license should be found.

-----------

The following npm package may be included in this product:

 - dep-three@1.0.0

This package contains the following license and notice below:

# Dep Three

This license file is spelt \`LICENSE\`.
This license should be found.

-----------

The following npm packages may be included in this product:

 - dep-two-duplicate@1.0.0
 - dep-two@1.0.0

These packages each contain the following license and notice below:

# Dep Two

This license file is spelt \`LICENCE.md\`.
This license should be found.

-----------

The following npm package may be included in this product:

 - dep-five@1.0.0

This package contains the following license and notice below:

MIT

-----------

This file was generated with the generate-license-file npm package!
https://www.npmjs.com/package/generate-license-file
"
`;

exports[`cli for a relative path no-spinner should match snapshot when --no-spinner is given 1`] = `
"This file was generated with the generate-license-file npm package!
https://www.npmjs.com/package/generate-license-file

The following npm package may be included in this product:

 - dep-four@1.0.0

This package contains the following license and notice below:

# Dep Four

This license file is spelt \`LICENCE\`.
This license should be found.

-----------

The following npm package may be included in this product:

 - dep-one@1.0.0

This package contains the following license and notice below:

# Dep One

This license file is spelt \`LICENSE.md\`.
This license should be found.

-----------

The following npm package may be included in this product:

 - dep-three@1.0.0

This package contains the following license and notice below:

# Dep Three

This license file is spelt \`LICENSE\`.
This license should be found.

-----------

The following npm packages may be included in this product:

 - dep-two-duplicate@1.0.0
 - dep-two@1.0.0

These packages each contain the following license and notice below:

# Dep Two

This license file is spelt \`LICENCE.md\`.
This license should be found.

-----------

The following npm package may be included in this product:

 - dep-five@1.0.0

This package contains the following license and notice below:

MIT

-----------

This file was generated with the generate-license-file npm package!
https://www.npmjs.com/package/generate-license-file
"
`;

exports[`cli for a relative path omit-versions should match snapshot when --omit-versions is given 1`] = `
"This file was generated with the generate-license-file npm package!
https://www.npmjs.com/package/generate-license-file

The following npm package may be included in this product:

 - dep-four

This package contains the following license and notice below:

# Dep Four

This license file is spelt \`LICENCE\`.
This license should be found.

-----------

The following npm package may be included in this product:

 - dep-one

This package contains the following license and notice below:

# Dep One

This license file is spelt \`LICENSE.md\`.
This license should be found.

-----------

The following npm package may be included in this product:

 - dep-three

This package contains the following license and notice below:

# Dep Three

This license file is spelt \`LICENSE\`.
This license should be found.

-----------

The following npm packages may be included in this product:

 - dep-two
 - dep-two-duplicate

These packages each contain the following license and notice below:

# Dep Two

This license file is spelt \`LICENCE.md\`.
This license should be found.

-----------

The following npm package may be included in this product:

 - dep-five

This package contains the following license and notice below:

MIT

-----------

This file was generated with the generate-license-file npm package!
https://www.npmjs.com/package/generate-license-file
"
`;

exports[`cli for a relative path should match snapshot using -i and -o 1`] = `
"This file was generated with the generate-license-file npm package!
https://www.npmjs.com/package/generate-license-file

The following npm package may be included in this product:

 - dep-four@1.0.0

This package contains the following license and notice below:

# Dep Four

This license file is spelt \`LICENCE\`.
This license should be found.

-----------

The following npm package may be included in this product:

 - dep-one@1.0.0

This package contains the following license and notice below:

# Dep One

This license file is spelt \`LICENSE.md\`.
This license should be found.

-----------

The following npm package may be included in this product:

 - dep-three@1.0.0

This package contains the following license and notice below:

# Dep Three

This license file is spelt \`LICENSE\`.
This license should be found.

-----------

The following npm packages may be included in this product:

 - dep-two-duplicate@1.0.0
 - dep-two@1.0.0

These packages each contain the following license and notice below:

# Dep Two

This license file is spelt \`LICENCE.md\`.
This license should be found.

-----------

The following npm package may be included in this product:

 - dep-five@1.0.0

This package contains the following license and notice below:

MIT

-----------

This file was generated with the generate-license-file npm package!
https://www.npmjs.com/package/generate-license-file
"
`;

exports[`cli for a relative path should match snapshot with --input and --output wrapped in double quotes 1`] = `
"This file was generated with the generate-license-file npm package!
https://www.npmjs.com/package/generate-license-file

The following npm package may be included in this product:

 - dep-four@1.0.0

This package contains the following license and notice below:

# Dep Four

This license file is spelt \`LICENCE\`.
This license should be found.

-----------

The following npm package may be included in this product:

 - dep-one@1.0.0

This package contains the following license and notice below:

# Dep One

This license file is spelt \`LICENSE.md\`.
This license should be found.

-----------

The following npm package may be included in this product:

 - dep-three@1.0.0

This package contains the following license and notice below:

# Dep Three

This license file is spelt \`LICENSE\`.
This license should be found.

-----------

The following npm packages may be included in this product:

 - dep-two-duplicate@1.0.0
 - dep-two@1.0.0

These packages each contain the following license and notice below:

# Dep Two

This license file is spelt \`LICENCE.md\`.
This license should be found.

-----------

The following npm package may be included in this product:

 - dep-five@1.0.0

This package contains the following license and notice below:

MIT

-----------

This file was generated with the generate-license-file npm package!
https://www.npmjs.com/package/generate-license-file
"
`;

exports[`cli for a relative path should match snapshot with --output given before --input 1`] = `
"This file was generated with the generate-license-file npm package!
https://www.npmjs.com/package/generate-license-file

The following npm package may be included in this product:

 - dep-four@1.0.0

This package contains the following license and notice below:

# Dep Four

This license file is spelt \`LICENCE\`.
This license should be found.

-----------

The following npm package may be included in this product:

 - dep-one@1.0.0

This package contains the following license and notice below:

# Dep One

This license file is spelt \`LICENSE.md\`.
This license should be found.

-----------

The following npm package may be included in this product:

 - dep-three@1.0.0

This package contains the following license and notice below:

# Dep Three

This license file is spelt \`LICENSE\`.
This license should be found.

-----------

The following npm packages may be included in this product:

 - dep-two-duplicate@1.0.0
 - dep-two@1.0.0

These packages each contain the following license and notice below:

# Dep Two

This license file is spelt \`LICENCE.md\`.
This license should be found.

-----------

The following npm package may be included in this product:

 - dep-five@1.0.0

This package contains the following license and notice below:

MIT

-----------

This file was generated with the generate-license-file npm package!
https://www.npmjs.com/package/generate-license-file
"
`;

exports[`cli for a relative path should match snapshot with just --input and --output 1`] = `
"This file was generated with the generate-license-file npm package!
https://www.npmjs.com/package/generate-license-file

The following npm package may be included in this product:

 - dep-four@1.0.0

This package contains the following license and notice below:

# Dep Four

This license file is spelt \`LICENCE\`.
This license should be found.

-----------

The following npm package may be included in this product:

 - dep-one@1.0.0

This package contains the following license and notice below:

# Dep One

This license file is spelt \`LICENSE.md\`.
This license should be found.

-----------

The following npm package may be included in this product:

 - dep-three@1.0.0

This package contains the following license and notice below:

# Dep Three

This license file is spelt \`LICENSE\`.
This license should be found.

-----------

The following npm packages may be included in this product:

 - dep-two-duplicate@1.0.0
 - dep-two@1.0.0

These packages each contain the following license and notice below:

# Dep Two

This license file is spelt \`LICENCE.md\`.
This license should be found.

-----------

The following npm package may be included in this product:

 - dep-five@1.0.0

This package contains the following license and notice below:

MIT

-----------

This file was generated with the generate-license-file npm package!
https://www.npmjs.com/package/generate-license-file
"
`;

exports[`cli for an absolute path ci should match snapshot when --ci is given 1`] = `
"This file was generated with the generate-license-file npm package!
https://www.npmjs.com/package/generate-license-file

The following npm package may be included in this product:

 - dep-four@1.0.0

This package contains the following license and notice below:

# Dep Four

This license file is spelt \`LICENCE\`.
This license should be found.

-----------

The following npm package may be included in this product:

 - dep-one@1.0.0

This package contains the following license and notice below:

# Dep One

This license file is spelt \`LICENSE.md\`.
This license should be found.

-----------

The following npm package may be included in this product:

 - dep-three@1.0.0

This package contains the following license and notice below:

# Dep Three

This license file is spelt \`LICENSE\`.
This license should be found.

-----------

The following npm packages may be included in this product:

 - dep-two-duplicate@1.0.0
 - dep-two@1.0.0

These packages each contain the following license and notice below:

# Dep Two

This license file is spelt \`LICENCE.md\`.
This license should be found.

-----------

The following npm package may be included in this product:

 - dep-five@1.0.0

This package contains the following license and notice below:

MIT

-----------

This file was generated with the generate-license-file npm package!
https://www.npmjs.com/package/generate-license-file
"
`;

exports[`cli for an absolute path for line ending crlf should match snapshot when --eol is given 1`] = `
"This file was generated with the generate-license-file npm package!
https://www.npmjs.com/package/generate-license-file

The following npm package may be included in this product:

 - dep-four@1.0.0

This package contains the following license and notice below:

# Dep Four

This license file is spelt \`LICENCE\`.
This license should be found.

-----------

The following npm package may be included in this product:

 - dep-one@1.0.0

This package contains the following license and notice below:

# Dep One

This license file is spelt \`LICENSE.md\`.
This license should be found.

-----------

The following npm package may be included in this product:

 - dep-three@1.0.0

This package contains the following license and notice below:

# Dep Three

This license file is spelt \`LICENSE\`.
This license should be found.

-----------

The following npm packages may be included in this product:

 - dep-two-duplicate@1.0.0
 - dep-two@1.0.0

These packages each contain the following license and notice below:

# Dep Two

This license file is spelt \`LICENCE.md\`.
This license should be found.

-----------

The following npm package may be included in this product:

 - dep-five@1.0.0

This package contains the following license and notice below:

MIT

-----------

This file was generated with the generate-license-file npm package!
https://www.npmjs.com/package/generate-license-file
"
`;

exports[`cli for an absolute path for line ending lf should match snapshot when --eol is given 1`] = `
"This file was generated with the generate-license-file npm package!
https://www.npmjs.com/package/generate-license-file

The following npm package may be included in this product:

 - dep-four@1.0.0

This package contains the following license and notice below:

# Dep Four

This license file is spelt \`LICENCE\`.
This license should be found.

-----------

The following npm package may be included in this product:

 - dep-one@1.0.0

This package contains the following license and notice below:

# Dep One

This license file is spelt \`LICENSE.md\`.
This license should be found.

-----------

The following npm package may be included in this product:

 - dep-three@1.0.0

This package contains the following license and notice below:

# Dep Three

This license file is spelt \`LICENSE\`.
This license should be found.

-----------

The following npm packages may be included in this product:

 - dep-two-duplicate@1.0.0
 - dep-two@1.0.0

These packages each contain the following license and notice below:

# Dep Two

This license file is spelt \`LICENCE.md\`.
This license should be found.

-----------

The following npm package may be included in this product:

 - dep-five@1.0.0

This package contains the following license and notice below:

MIT

-----------

This file was generated with the generate-license-file npm package!
https://www.npmjs.com/package/generate-license-file
"
`;

exports[`cli for an absolute path no-spinner should match snapshot when --no-spinner is given 1`] = `
"This file was generated with the generate-license-file npm package!
https://www.npmjs.com/package/generate-license-file

The following npm package may be included in this product:

 - dep-four@1.0.0

This package contains the following license and notice below:

# Dep Four

This license file is spelt \`LICENCE\`.
This license should be found.

-----------

The following npm package may be included in this product:

 - dep-one@1.0.0

This package contains the following license and notice below:

# Dep One

This license file is spelt \`LICENSE.md\`.
This license should be found.

-----------

The following npm package may be included in this product:

 - dep-three@1.0.0

This package contains the following license and notice below:

# Dep Three

This license file is spelt \`LICENSE\`.
This license should be found.

-----------

The following npm packages may be included in this product:

 - dep-two-duplicate@1.0.0
 - dep-two@1.0.0

These packages each contain the following license and notice below:

# Dep Two

This license file is spelt \`LICENCE.md\`.
This license should be found.

-----------

The following npm package may be included in this product:

 - dep-five@1.0.0

This package contains the following license and notice below:

MIT

-----------

This file was generated with the generate-license-file npm package!
https://www.npmjs.com/package/generate-license-file
"
`;

exports[`cli for an absolute path omit-versions should match snapshot when --omit-versions is given 1`] = `
"This file was generated with the generate-license-file npm package!
https://www.npmjs.com/package/generate-license-file

The following npm package may be included in this product:

 - dep-four

This package contains the following license and notice below:

# Dep Four

This license file is spelt \`LICENCE\`.
This license should be found.

-----------

The following npm package may be included in this product:

 - dep-one

This package contains the following license and notice below:

# Dep One

This license file is spelt \`LICENSE.md\`.
This license should be found.

-----------

The following npm package may be included in this product:

 - dep-three

This package contains the following license and notice below:

# Dep Three

This license file is spelt \`LICENSE\`.
This license should be found.

-----------

The following npm packages may be included in this product:

 - dep-two
 - dep-two-duplicate

These packages each contain the following license and notice below:

# Dep Two

This license file is spelt \`LICENCE.md\`.
This license should be found.

-----------

The following npm package may be included in this product:

 - dep-five

This package contains the following license and notice below:

MIT

-----------

This file was generated with the generate-license-file npm package!
https://www.npmjs.com/package/generate-license-file
"
`;

exports[`cli for an absolute path should match snapshot using -i and -o 1`] = `
"This file was generated with the generate-license-file npm package!
https://www.npmjs.com/package/generate-license-file

The following npm package may be included in this product:

 - dep-four@1.0.0

This package contains the following license and notice below:

# Dep Four

This license file is spelt \`LICENCE\`.
This license should be found.

-----------

The following npm package may be included in this product:

 - dep-one@1.0.0

This package contains the following license and notice below:

# Dep One

This license file is spelt \`LICENSE.md\`.
This license should be found.

-----------

The following npm package may be included in this product:

 - dep-three@1.0.0

This package contains the following license and notice below:

# Dep Three

This license file is spelt \`LICENSE\`.
This license should be found.

-----------

The following npm packages may be included in this product:

 - dep-two-duplicate@1.0.0
 - dep-two@1.0.0

These packages each contain the following license and notice below:

# Dep Two

This license file is spelt \`LICENCE.md\`.
This license should be found.

-----------

The following npm package may be included in this product:

 - dep-five@1.0.0

This package contains the following license and notice below:

MIT

-----------

This file was generated with the generate-license-file npm package!
https://www.npmjs.com/package/generate-license-file
"
`;

exports[`cli for an absolute path should match snapshot with --input and --output wrapped in double quotes 1`] = `
"This file was generated with the generate-license-file npm package!
https://www.npmjs.com/package/generate-license-file

The following npm package may be included in this product:

 - dep-four@1.0.0

This package contains the following license and notice below:

# Dep Four

This license file is spelt \`LICENCE\`.
This license should be found.

-----------

The following npm package may be included in this product:

 - dep-one@1.0.0

This package contains the following license and notice below:

# Dep One

This license file is spelt \`LICENSE.md\`.
This license should be found.

-----------

The following npm package may be included in this product:

 - dep-three@1.0.0

This package contains the following license and notice below:

# Dep Three

This license file is spelt \`LICENSE\`.
This license should be found.

-----------

The following npm packages may be included in this product:

 - dep-two-duplicate@1.0.0
 - dep-two@1.0.0

These packages each contain the following license and notice below:

# Dep Two

This license file is spelt \`LICENCE.md\`.
This license should be found.

-----------

The following npm package may be included in this product:

 - dep-five@1.0.0

This package contains the following license and notice below:

MIT

-----------

This file was generated with the generate-license-file npm package!
https://www.npmjs.com/package/generate-license-file
"
`;

exports[`cli for an absolute path should match snapshot with --output given before --input 1`] = `
"This file was generated with the generate-license-file npm package!
https://www.npmjs.com/package/generate-license-file

The following npm package may be included in this product:

 - dep-four@1.0.0

This package contains the following license and notice below:

# Dep Four

This license file is spelt \`LICENCE\`.
This license should be found.

-----------

The following npm package may be included in this product:

 - dep-one@1.0.0

This package contains the following license and notice below:

# Dep One

This license file is spelt \`LICENSE.md\`.
This license should be found.

-----------

The following npm package may be included in this product:

 - dep-three@1.0.0

This package contains the following license and notice below:

# Dep Three

This license file is spelt \`LICENSE\`.
This license should be found.

-----------

The following npm packages may be included in this product:

 - dep-two-duplicate@1.0.0
 - dep-two@1.0.0

These packages each contain the following license and notice below:

# Dep Two

This license file is spelt \`LICENCE.md\`.
This license should be found.

-----------

The following npm package may be included in this product:

 - dep-five@1.0.0

This package contains the following license and notice below:

MIT

-----------

This file was generated with the generate-license-file npm package!
https://www.npmjs.com/package/generate-license-file
"
`;

exports[`cli for an absolute path should match snapshot with just --input and --output 1`] = `
"This file was generated with the generate-license-file npm package!
https://www.npmjs.com/package/generate-license-file

The following npm package may be included in this product:

 - dep-four@1.0.0

This package contains the following license and notice below:

# Dep Four

This license file is spelt \`LICENCE\`.
This license should be found.

-----------

The following npm package may be included in this product:

 - dep-one@1.0.0

This package contains the following license and notice below:

# Dep One

This license file is spelt \`LICENSE.md\`.
This license should be found.

-----------

The following npm package may be included in this product:

 - dep-three@1.0.0

This package contains the following license and notice below:

# Dep Three

This license file is spelt \`LICENSE\`.
This license should be found.

-----------

The following npm packages may be included in this product:

 - dep-two-duplicate@1.0.0
 - dep-two@1.0.0

These packages each contain the following license and notice below:

# Dep Two

This license file is spelt \`LICENCE.md\`.
This license should be found.

-----------

The following npm package may be included in this product:

 - dep-five@1.0.0

This package contains the following license and notice below:

MIT

-----------

This file was generated with the generate-license-file npm package!
https://www.npmjs.com/package/generate-license-file
"
`;
